src/mscorlib/src/System/Enum.cs

```diff
@@ -415,47 +418,53 @@ private static bool TryParseEnum(Type enumType, String value, bool ignoreCase, r
                 }
             }
 
-            String[] values = value.Split(enumSeperatorCharArray);
-
-            // Find the field.Lets assume that these are always static classes because the class is
-            //  an enum.
+            // Find the field. Let's assume that these are always static classes 
+            // because the class is an enum.
             ValuesAndNames entry = GetCachedValuesAndNames(rtType, true);
-
             String[] enumNames = entry.Names;
             ulong[] enumValues = entry.Values;
-            
-            for (int i = 0; i < values.Length; i++)
+
+            StringComparison comparison = ignoreCase ? 
+                StringComparison.OrdinalIgnoreCase : 
+                StringComparison.Ordinal;
+
+            for (int valueIndex = 0; valueIndex <= value.Length; valueIndex++)
             {
-                values[i] = values[i].Trim(); // We need to remove whitespace characters
+                // Find the next separator, if there is one, otherwise the end of the string.
+                int endIndex = value.IndexOf(enumSeparatorChar, valueIndex);
+                if (endIndex == -1)
+                {
+                    endIndex = value.Length;
+                }
 
-                bool success = false;
+                // Shift the starting and ending indices to eliminate whitespace
+                int endIndexNoWhitespace = endIndex;
+                while (valueIndex < endIndex && Char.IsWhiteSpace(value[valueIndex])) valueIndex++;
+                while (endIndexNoWhitespace > valueIndex && Char.IsWhiteSpace(value[endIndexNoWhitespace - 1])) endIndexNoWhitespace--;
+                int valueSubstringLength = endIndexNoWhitespace - valueIndex;
 
-                for (int j = 0; j < enumNames.Length; j++)
+                // Try to match this substring against each enum name
+                bool success = false;
+                for (int i = 0; i < enumNames.Length; i++)
                 {
-                    if (ignoreCase)
-                    {
-                        if (String.Compare(enumNames[j], values[i], StringComparison.OrdinalIgnoreCase) != 0)
-                            continue;
-                    }
-                    else
+                    if (string.Compare(enumNames[i], 0, value, valueIndex, Math.Max(enumNames[i].Length, valueSubstringLength), comparison) == 0)
```

Comment:
Check for `enumNames[i].Length == valueSubstringLength` should be better than doing `Math.Max` (faster; and easier to see that it is correct - no comparing random stuff from `valueSubstring`),